test/powershell/Modules/Microsoft.PowerShell.Management/Set-Location.Tests.ps1

```diff
@@ -53,6 +53,22 @@ Describe "Set-Location" -Tags "CI" {
         $(Get-Location).Path | Should -BeExactly $testPath.FullName
     }
 
+    It "Should not use filesystem root folder if not in filesystem provider" -Skip:(!$IsWindows) {
+        # find filesystem root folder that doesn't exist in HKCU:
+        $foundFolder = $false
+        foreach ($folder in (Get-ChildItem "${env:SystemDrive}\" -Directory)) {
+            if (!(Test-Path "HKCU:\$($folder.Name)")) {
```

Comment:
I think the convention is to use `-not` instead of `!` simply because it's more visible.